src/python/WMCore/MicroService/MSTransferor/MSTransferor.py

```diff
@@ -676,3 +720,58 @@ def _diskPNNs(self, pnnList):
             else:
                 diskPNNs.add(pnn)
         return diskPNNs
+
+    def updateSites(self, rec):
+        """
+        Update sites API provides asynchronous update of site list information
+        :param rec: JSON payload with the following data structures: {'workflow': <wflow name>}
+        :return: either empty list (no errors) or list of errors
+        """
+        # preserve provided payload to local file system
+        wflowName = rec['workflow']
+        status = self.updateStorage(wflowName)
+        if status == 'ok':
+            return []
+        err = MSTransferorStorageError(rec)
+        self.logger.error(err)
+        return [err.error()]
+
+    def updateStorage(self, wflowName):
+        """
+        Save workflow data to persistent storage
+        :param wflowName: name of the workflow
+        :return: status of this operation
+        """
+        try:
+            fname = '{}/{}'.format(self.storage, wflowName)
+            with open(fname, 'w', encoding="utf-8"):
+                # we perform touch operation on file system, i.e. create empty file
+                self.logger.info("for workflow %s add entry to persistent storage", wflowName)
+                os.utime(fname, None)
+            return 'ok'
+        except Exception as exp:
+            msg = "Unable to save workflow '%s' to storage=%s. Error: %s", wflowName, self.storage, str(exp)
+            self.logger.exception(msg)
+            return str(exp)
+
+    def checkDataReplacement(self, wflow):
+        """
+        Check if given workflow exists on local storage and set dataReplacement flag if it is the case
+        :param wflow: workflow object
+        :return: nothing
+        """
+        fname = '{}/{}'.format(self.storage, wflow.getName())
+        if os.path.exists(fname):
+            self.logger.info("for workflow %s will perform data replacement", wflow.getName())
+            wflow.dataReplacement = True
+
+    def cleanupStorage(self, wflowName):
+        """
+        Remove workflow from persistent storage
+        :param wflowName: name of workflow
+        :return: nothing
+        """
+        fname = '{}/{}'.format(self.storage, wflowName)
```

Comment:
To avoid potential double slash `//`, don't you think `os.path.join()` would be safer here (and in the other places you are concatenating file to path)?

Reply:
ok